### ISSUE TITLE
error: Invalid usage of publishCreate() :: Values must have an `undefined`, instead got

### ISSUE THREAD

**Issue by chipsenkbeil** [balderdashy/sails#3350]
My package.json is set up to pull the latest code (there are some fixes that went in past 0.12.0-rc3 and I have to use 0.12.0 due to some issues with `sails-auth`):

```json
"dependencies": {
    ...
    "sails": "balderdashy/sails",
    ...
}
```

I created a new model and controller using the standard sails generator. The controller is empty, but I've added the following schema to the new model (called Card):

```node
module.exports = {
  attributes: {
    id: {
      type: "string",
      required: true,
      index: true,
      notNull: true,
      unique: true
    },
    type: {
      type: "string",
      defaultsTo: "unknown"
    },
    title: {
      type: "string",
      defaultsTo: "unknown"
    },
    description: {
      type: "string",
      defaultsTo: ""
    },
    data: {
      type: "json",
      required: true
    },
    append: {
      type: "boolean",
      defaultsTo: false
    }
  }
};
```

Visiting this endpoint via GET `https://localhost:3000/card/create?id=test&data="{'key': 3}"`, I see the following in my server's logs.

```js
error: Invalid usage of publishCreate() :: Values must have an `undefined`, instead got ::
{ id: 'test',
  data: '{\'key\': 3}',
  type: 'unknown',
  title: 'unknown',
  description: '',
  append: false,
  createdAt: '2015-10-30T14:36:40.591Z',
  updatedAt: '2015-10-30T14:36:40.591Z' }
```

The card itself does get created, though. Unlike other issues related to this, I get `undefined` as the needed value, so I have no idea what is wrong. For the data, I've used various combinations of single and double quotes to no avail.

I'm using the local disk database for now as well.

**Comment by sailsbot** [comment]
Thanks for posting, @chipsenkbeil.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
